internal/sync/syncer.go

```diff
@@ -239,7 +253,31 @@ func (s *Syncer) runOnce(ctx context.Context, key string, worker int) error {
 		Namespace: gitSync.Namespace,
 		Name:      gitSync.Name,
 	}
-	return updateCommitStatus(ctx, s.client, log, namespacedName, commitHash, synced, syncMessage)
+
+	err = updateCommitStatus(ctx, s.client, log, namespacedName, commitHash, synced, syncMessage)
+	if err != nil {
```

Comment:
Apologize on this,  but even with errors we should check if the GitSync is deleted. Thus, I think the code should be,
```
if !gitSync.GetDeletionTimestamp().IsZero() {
   ...
} else {
   return updateCommitStatus(ctx, s.client, log, namespacedName, commitHash, synced, syncMessage)
}
```